scenarios/python/console/captioning/caption.py

```diff
@@ -0,0 +1,275 @@
+# Dependencies
+# Note: abc = abstract base classes
+from collections.abc import Mapping
+import datetime
+from os import linesep, remove
+from os.path import exists
+from pathlib import Path
+from sys import argv
+from time import sleep
+from typing import Optional
+
+# Notes:
+# - Install the Speech SDK. Run:
+# pip install azure-cognitiveservices-speech
+# - The Python Speech SDK on Windows requires the Microsoft Visual C++ Redistributable for Visual Studio 2015, 2017, 2019, or 2022 on the system. See:
+# https://docs.microsoft.com/azure/cognitive-services/speech-service/quickstarts/setup-platform?pivots=programming-language-python&tabs=dotnet%2Cwindows%2Cjre%2Cbrowser#prerequisites
+
+import azure.cognitiveservices.speech as speechsdk
+
+# Types
+
+# See
+# https://stackoverflow.com/a/28452633
+class Read_Only_Dict(Mapping):
+    def __init__(self, data):
+        self._data = data
+    def __getitem__(self, key): 
+        return self._data[key]
+    def __len__(self):
+        return len(self._data)
+    def __iter__(self):
+        return iter(self._data)
+
+class Too_Few_Arguments(Exception) :
+    pass
+
+# Helper functions
+
+def get_cmd_option(option : str) -> Optional[str] :
+    argc = len(argv)
+    if option in argv :
+        index = argv.index(option)
+        if index < argc - 2 :
+# We found the option (for example, "-o"), so advance from that to the value (for example, "filename").
+            return argv[index + 1]
+        else :
+            return None
+    else :
+        return None
+
+def cmd_option_exists(option : str) -> bool :
+    return option in argv
+
+def user_config_from_args() -> Read_Only_Dict :
+    argc = len(argv)
+    if argc < 3 :
```

Comment:
Our samples try to follow this Python pep8 convention for blank lines

[https://stackoverflow.com/questions/2953250/python-pep8-blank-lines-convention](https://stackoverflow.com/questions/2953250/python-pep8-blank-lines-convention)